lib/sequin_web/live/sink_consumers/show.ex

```diff
@@ -401,6 +403,52 @@ defmodule SequinWeb.SinkConsumersLive.Show do
     end
   end
 
+  def handle_event("pause_backfill", %{"backfill_id" => backfill_id}, socket) do
+    case Consumers.get_backfill(backfill_id) do
+      {:ok, backfill} ->
+        if backfill.sink_consumer_id == socket.assigns.consumer.id do
```

Comment:
This is good protection. Other thing you can do is just blow up if they don't match (this means either a (a) bug on our service or (b) malicious user

```suggestion
    # bind consumer_id earlier
    case Consumers.get_backfill(backfill_id) do
      {:ok, %{ consumer_id: ^consumer_id } ->
        if backfill.sink_consumer_id == socket.assigns.consumer.id do
```

Or we can add a new function `get_backfill_for_consumer(consumer_id, backfill_id)`

The reason I like these two approaches is that it's way clearer to the reader that the backfill_id _not_ belonging to the consumer is exceptional

Reply:
And then with a `with` clause this can become one happy path + one `else` fall-through